### PR TITLE
v0.101.0 release prep

### PR DESCRIPTION
**Pull request by cpu**
- [X] Updated README with changelog
- [x] Updated README with new version + date
- [x] Checked for outdated library dependencies (keep untrusted at 0.7 to match *ring*)
- [x] Tests pass locally with `--all-features` and `--no-default-features`
- [x] Updated `Cargo.toml` with new version
- [x] `cargo publish --dry-run` succeeds

### CODE REVIEW

**Comment by ctz** [comment]
This looks good. Any idea what is going on with the coverage measurement on this PR?

File: README.md [hunk 1]
```diff
@@ -54,6 +54,23 @@ For these tasks you may prefer using webpki in combination with libraries like
 Changelog
 =========
 
+* Next release:
+  - *Breaking change*: added `EndEntity::verify_is_valid_tls_client_cert`
+    argument for certificate revocation lists.
+  - Added certificate revocation list (CRL) support.
+  - *Breaking change*: removed deprecated `Time::try_from` fn.
+  - *Breaking change*: removed deprecated `DnsName::From<DnsNameRef<'_>>` trait.
```
Comment (djc) [inline README.md:62]: I find this way of describing trait impls confusing (and you wrote "removed .. trait", which is arguably not what we did). I would just write "`impl From<DnsNameRef<'_>> for DnsName`" or "`From<DnsNameRef<'_>` impl for `DnsName`". (Same for the line below.)

File: README.md [hunk 2]
```diff
@@ -54,6 +54,23 @@ For these tasks you may prefer using webpki in combination with libraries like
 Changelog
 =========
 
+* Next release:
+  - *Breaking change*: added `EndEntity::verify_is_valid_tls_client_cert`
+    argument for certificate revocation lists.
+  - Added certificate revocation list (CRL) support.
```
Comment (djc) [inline README.md:60]: I feel like we should either have this line first (as the headline feature of the release) or keep all the breaking changes first (as the thing most likely to impact downstream users), but avoid keeping this in between the breaking changes.

File: README.md [hunk 3]
```diff
@@ -54,6 +54,23 @@ For these tasks you may prefer using webpki in combination with libraries like
...
+  - *Breaking change*: removed deprecated `Time::try_from` fn.
+  - *Breaking change*: removed deprecated `DnsName::From<DnsNameRef<'_>>` trait.
+  - *Breaking change*: replaced `DnsNameRef::AsRef<[u8]>` with
+    `DnsNameRef::AsRef<str>`.
+  - Improved specificity of errors returned from
+    `EndEntityCert::verify_is_valid_tls_client_cert` and
+    `EndEntityCert::verify_is_valid_tls_server_cert`.
+  - Changed `EndEntityCert::verify_is_valid_for_subject_name` to ignore invalid
+    names when verifying cert is valid for a provided subject.
+  - MSRV increased to Rust 1.57.
```
Comment (djc) [inline README.md:70]: I would move the MSRV bump to the bottom, it's probably only relevant to a few users.

File: README.md [hunk 4]
```diff
@@ -54,6 +54,23 @@ For these tasks you may prefer using webpki in combination with libraries like
...
+  - Improved error specificity for malformed subject alternate names.
+  - Added `EndEntityCert::dns_names` method for returning a list of DNS subject
+    alternate names from an end entyt cert.
```
Comment (djc) [inline README.md:73]: "entyt" -> "entity"

Comment (cpu) [reply README.md]: Fixed. Thanks

File: README.md (hunk 3 above)
Comment (cpu) [reply README.md:72]: Sounds good. Changed.

File: README.md (hunk 2 above)
Comment (cpu) [reply README.md:63]: I re-arranged the items a little bit. Previously I wrote them more or less in the order they landed in-tree. Now they're written with breaking changes first, then the CRL work, then small misc changes, and the MSRV update last.

File: README.md (hunk 1 above)
Comment (cpu) [reply README.md]: PTAL at the latest revision. I hope I interpreted your comment correctly.

**Comment by cpu** [comment]
Navigating to the [actual codecov page](https://app.codecov.io/gh/rustls/webpki/pull/109) for the branch shows these are "indirect changes". It also mentions:
> Coverage data is based on HEAD [66cfa3a](https://app.codecov.io/gh/rustls/webpki/commit/66cfa3a89892a9a5a76bc3a39992ad7933e982b7) compared to BASE [1d1f05a](https://app.codecov.io/gh/rustls/webpki/commit/1d1f05a3c777365b2ebd4b36235cf001ca2e6cca)

The BASE rev there is stale and from Jun 23, 2023. It looks like all of the CRL work that landed since then didn't have coverage information uploaded and so we're seeing indirect changes associated with this branch.

**Comment by ctz** [comment]
I think https://github.com/rustls/webpki/actions/runs/5444991778/jobs/9903605334 (which is a build on main) failed silently to upload coverage, so  codecov still thinks main has its old coverage and attributes the change to this PR. I will look at adding a token to our GH action which apparently improves the reliability of the upload process.

Could we spread some of the CRL tests between Owned* and Borrowed* types? It looks like all the tests are against Borrowed* types.

**Comment by cpu** [comment]
> Could we spread some of the CRL tests between Owned* and Borrowed* types? It looks like all the tests are against Borrowed* types.

:+1: I'll add some coverage for the owned types in a separate PR.

**Comment by cpu** [comment]
>  I'll add some coverage for the owned types in a separate PR.

https://github.com/rustls/webpki/pull/111

**Review by djc — APPROVED** [review]

**Comment by cpu** [comment]
> Updated README with new version + date

I've left off this step until things are looking ready to go since I can't push the release myself and don't want to include the wrong date.

If folks are happy with the changelog and someone is available to publish the crate I can replace "* Next release:" with "* 0.101.0 (2023-07-04):" and merge.

**Comment by djc** [comment]
Feel free to just set the date to today, I will be available to publish if necessary.

**Comment by cpu** [comment]
@djc Thanks, appreciated. I updated w/ the date.

**Comment by cpu** [comment]
Published https://crates.io/crates/rustls-webpki

**Pull request merged**